src/components/UsersCountModal/index.tsx

```diff
@@ -0,0 +1,100 @@
+import classNames from 'classnames';
+import React, { ReactElement, useEffect, useState } from 'react';
+
+import { fetchLikersByPost, fetchUserFollowers, fetchUserFollowings } from '../../ducks/posts';
+
+import Modal, { ModalContent, ModalHeader } from '../Modal';
+import { UserRow } from '../DiscoverUserPanel';
+
+function UsersList(props: { onClose: () => void; users: string[]; title: string }): ReactElement {
+  const { onClose, title, users } = props;
+  return (
+    <Modal className={classNames('w-148')} onClose={onClose}>
+      <ModalHeader>{title}</ModalHeader>
+      <ModalContent>
+        {users.map(ens => (
+          <UserRow key={ens} name={ens} />
+        ))}
+      </ModalContent>
+    </Modal>
+  );
+}
+
+export enum Item {
+  Like = 'Like',
+  Follower = 'Follower',
+  Following = 'Following',
+}
+
+const fetch = {
+  [Item.Like]: fetchLikersByPost,
+  [Item.Follower]: fetchUserFollowers,
+  [Item.Following]: fetchUserFollowings,
+};
+
+function MaybePlural(props: { users: string[]; text: Item }): ReactElement {
+  const { users, text } = props;
+  const count = users.length;
+  const many = count > 1 && text !== Item.Following;
+
+  return (
+    <div className="flex flex-row flex-nowrap item-center">
+      <div className="font-semibold">{count}</div>
+      <div className="ml-2 text-gray-500">{`${text}${many ? 's' : ''}`}</div>
+    </div>
+  );
+}
+
+export default function UsersCountModal(props: {
+  className: string;
+  item: Item;
+  id: string;
+}): ReactElement {
+  const { className, item, id } = props;
+  const [showingList, setShowList] = useState(false);
+
+  const [users, setItems] = useState<string[] | null>(null);
+  const count = users?.length;
+  const many = count && count > 1 && item !== Item.Following;
+  const text = `${item}${many ? 's' : ''}`;
+  const title = item === Item.Like ? 'Liked By' : text;
+
+  useEffect(() => {
+    setShowList(false);
+
+    fetch[item](id).then(users => setItems(users));
+  }, [id]);
+
+  console.log('debug', { showingList });
+  /*
+    if 0 likes show nothing
+    if 0 follow-er/ing, show "0 Follow-er/ing"
+   */
+  return users ? (
+    <>
+      <div className="flex flex-row flex-nowrap items-center text-light">
+        <div className={classNames(className)}>
+          <div className="hover:underline cursor-pointer" onClick={() => setShowList(true)}>
```

Comment:
I am not sure I got the styling right, the underlining is not continuous
![image](https://user-images.githubusercontent.com/38692952/196716581-1fc5071d-db43-4815-b608-d65185de0758.png)